iocore/net/SSL.h

```diff
@@ -0,0 +1,33 @@
+/** @file
+
+  Absorber of SSL library versions
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include <openssl/evp.h>
```

Comment:
SSL.h that doesn't include openssl/ssl.h would confuse me in the future.

digest.h would be needed if you used BoringSSL
https://github.com/google/boringssl/blob/f109f2087349712d3ac717d15fab48e130618110/include/openssl/digest.h#L81-L83

I may want to have this file as "tscore/ink_evp.h" if the name convention is still valid. In this case, the function name should probably be ink_evp_shax or something like it.

If you were not going to make any of these changes above, I'd suggest have this in the cc file that really needs it.

Reply:
My intention of this file is collecting ifdef directives comes from support different versions of SSL library or multiple SSL libraries. Because there are same ifdefs directives in many files.

I agree with this macro should be renamed, because this is exposed in global namespace. ( I prefer `ts_` prefix though ). 

I'll open another PR to introduce this file.

Reply:
I put back `evp_md_func` in .cc files for this PR.